app/assets/javascripts/controllers/cloud_network/cloud_network_form_controller.js

```diff
@@ -1,63 +1,68 @@
-      $scope.cloudNetworkModel.qos_policy_id = data.qos_policy_id;
-      $scope.cloudNetworkModel.shared = data.shared;
-      $scope.cloudNetworkModel.vlan_transparent = data.vlan_transparent;
-      $scope.modelCopy = angular.copy( $scope.cloudNetworkModel );
+ManageIQ.angular.app.controller('cloudNetworkFormController', ['cloudNetworkFormId', 'miqService', 'API', function(cloudNetworkFormId, miqService, API) {
+  var vm = this;
+
+  var init = function() {
+    vm.cloudNetworkModel = { name: '', ems_id: '', cloud_tenant_id: '' };
+    vm.formId = cloudNetworkFormId;
+    vm.model = "cloudNetworkModel";
+    vm.newRecord = cloudNetworkFormId === "new";
+    vm.saveable = miqService.saveable;
+
+    if (vm.newRecord) {
+      vm.cloudNetworkModel.name = "";
+      vm.cloudNetworkModel.enabled = true;
+      vm.cloudNetworkModel.external_facing = false;
+      vm.cloudNetworkModel.shared = false;
+      vm.cloudNetworkModel.vlan_transparent = false;
+      vm.afterGet = true;
+      vm.modelCopy = angular.copy( vm.cloudNetworkModel );
+    } else {
+      miqService.sparkleOn();
+      API.get("/api/cloud_networks/" + cloudNetworkFormId + "?attributes=cloud_tenant").then(function(data) {
+        vm.cloudNetworkModel.name = data.name;
+        vm.cloudNetworkModel.cloud_tenant_name = data.cloud_tenant.name;
+        vm.cloudNetworkModel.enabled = data.enabled;
+        vm.cloudNetworkModel.external_facing = data.external_facing;
+        vm.cloudNetworkModel.port_security_enabled = data.port_security_enabled;
+        vm.cloudNetworkModel.provider_network_type = data.provider_network_type;
+        vm.cloudNetworkModel.qos_policy_id = data.qos_policy_id;
+        vm.cloudNetworkModel.shared = data.shared;
+        vm.cloudNetworkModel.vlan_transparent = data.vlan_transparent;
+        vm.afterGet = true;
+        vm.modelCopy = angular.copy( vm.cloudNetworkModel );
+      }).catch(miqService.handleFailure);
       miqService.sparkleOff();
-    }).catch(miqService.handleFailure);
-  }
+    }
+  };
 
-  $scope.addClicked = function() {
+  vm.addClicked = function() {
     var url = 'create/new' + '?button=add';
-    miqService.miqAjaxButton(url, $scope.cloudNetworkModel, { complete: false });
+    miqService.miqAjaxButton(url, vm.cloudNetworkModel, { complete: false });
   };
 
-  $scope.cancelClicked = function() {
-    if (cloudNetworkFormId == 'new') {
+  vm.cancelClicked = function() {
+    if (vm.newRecord) {
       var url = '/cloud_network/create/new' + '?button=cancel';
     } else {
       var url = '/cloud_network/update/' + cloudNetworkFormId + '?button=cancel';
     }
     miqService.miqAjaxButton(url);
   };
 
-  $scope.saveClicked = function() {
+  vm.saveClicked = function() {
     var url = '/cloud_network/update/' + cloudNetworkFormId + '?button=save';
-    miqService.miqAjaxButton(url, $scope.cloudNetworkModel, { complete: false });
+    miqService.miqAjaxButton(url, vm.cloudNetworkModel, { complete: false });
```

Comment:
@AparnaKarve, no worries, that's a very valid question but unfortunately the answer is the same for all NetworkProvider driven items: All modifications are made behind the inventory curtain using raw_<commands> tapping directly into Openstack API.